lbrynet/lbrynet_daemon/LBRYDaemon.py

```diff
@@ -1904,34 +1922,20 @@ def jsonrpc_search_nametrie(self, p):
         else:
             return self._render_response(None, BAD_REQUEST)
 
+        # TODO: have ui accept the actual outputs
         def _clean(n):
             t = []
             for i in n:
-                if i[0]:
-                    tr = {}
-                    tr.update(i[1][0]['value'])
-                    thumb = tr.get('thumbnail', None)
-                    if thumb is None:
-                        tr['thumbnail'] = "img/Free-speech-flag.svg"
-                    tr['name'] = i[1][0]['name']
-                    tr['cost_est'] = i[1][1]
-                    t.append(tr)
+                td = {k: i['value'][k] for k in i['value']}
+                td['cost_est'] = float(i['cost'])
+                td['thumbnail'] = i['value'].get('thumbnail', "img/Free-speech-flag.svg")
+                td['name'] = i['name']
+                t.append(td)
```

Comment:
Yes - lighthouse uses download_descriptor to cache sd blobs, which it then uses to calculate the stream costs. Previously, the daemon would calculate costs after getting the search results, now they're included in those results.

The (not very) _clean function was just to map the search return to the keys the ui uses. I think Alex has the UI querying lighthouse directly, and not through the daemon. In any case, when the UI handles the return keys 'cost', 'available', 'value', and 'name' then the _clean function can be deleted.